tests/PHPUnit/System/SpecialDateParamsTest.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link    https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Tests\System;
+
+use Piwik\Tests\Fixtures\VisitsInCurrentYear;
+use Piwik\Tests\Framework\TestCase\SystemTestCase;
+
+/**
+ * @group SpecialDateParams
+ * @group Core
+ */
+class SpecialDateParamsTest extends SystemTestCase
+{
+    public static $fixture = null; // initialized below class definition
+
+    /**
+     * @dataProvider getApiForTesting
+     */
+    public function testApi($api, $params)
+    {
+        $this->runApiTests($api, $params);
+    }
+
+    public function getApiForTesting()
+    {
+        return [
+            [
+                ['VisitsSummary.getVisits', 'Actions.getPageTitles'],
+                [
+                    'idSite'       => self::$fixture->idSite,
+                    'date'         => 'last week',
+                    'periods'      => ['day'],
+                    'testSuffix'   => '_lastweek',
+                ],
+            ],
+            [
+                ['VisitsSummary.getVisits', 'Actions.getPageTitles'],
+                [
+                    'idSite'       => self::$fixture->idSite,
+                    'date'         => 'last month',
+                    'periods'      => ['day'],
+                    'testSuffix'   => '_lastmonth',
+                ],
+            ],
+            [
+                ['VisitsSummary.getVisits', 'Actions.getPageTitles'],
+                [
+                    'idSite'       => self::$fixture->idSite,
+                    'date'         => 'last year',
+                    'periods'      => ['day'],
+                    'testSuffix'   => '_lastyear',
+                ],
+            ],
+            [
+                ['VisitsSummary.getVisits', 'Actions.getPageTitles'],
+                [
+                    'idSite'       => self::$fixture->idSite,
+                    'date'         => 'last year,last week',
+                    'periods'      => ['range'],
+                    'testSuffix'   => '_range',
+                ],
```

Comment:
Just for completeness, maybe we could change a couple of these dates to `last-week/lastYear`?

Reply:
done